packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerCloudState.ts

```diff
@@ -0,0 +1,116 @@
+              LayerCfnLogicalNamePrefix.LambdaLayerVersionPermission,
+              '',
+            ).replace(layerLogicalIdSuffix, '');
+
+            const accountIds = [];
+            const orgIds = [];
+            if (permissionTypeString === PermissionEnum.Private || permissionTypeString.startsWith(LegacyPermissionEnum.Private)) {
+              layerVersion.permissions.push({ type: PermissionEnum.Private });
+            } else if (permissionTypeString === PermissionEnum.Public || permissionTypeString.startsWith(LegacyPermissionEnum.Public)) {
+              layerVersion.permissions.push({ type: PermissionEnum.Public });
+            } else if (permissionTypeString.startsWith(PermissionEnum.AwsAccounts)) {
+              accountIds.push(permissionTypeString.replace(PermissionEnum.AwsAccounts, '').replace(`Legacy${layerVersion.Version}`, ''));
+            } else if (permissionTypeString.startsWith(LegacyPermissionEnum.AwsAccounts)) {
+              accountIds.push(permissionTypeString.replace(LegacyPermissionEnum.AwsAccounts, '').substring(0, 12));
+            } else if (permissionTypeString.startsWith(PermissionEnum.AwsOrg)) {
+              const orgId = permissionTypeString.replace(`${PermissionEnum.AwsOrg}o`, 'o-').replace(`Legacy${layerVersion.Version}`, '');
+              orgIds.push(orgId);
+            } else if (permissionTypeString.startsWith(LegacyPermissionEnum.AwsOrg)) {
+              const suffix = `${layerVersion.Version}`;
+              const orgId = permissionTypeString.replace(`${LegacyPermissionEnum.AwsOrg}o`, 'o-').slice(0, -1 * suffix.length);
+              orgIds.push(orgId);
+            }
+
+            if (accountIds.length > 0) {
+              layerVersion.permissions.push({
+                type: PermissionEnum.AwsAccounts,
+                accounts: accountIds,
+              });
+            }
+            if (orgIds.length > 0) {
+              layerVersion.permissions.push({
+                type: PermissionEnum.AwsOrg,
+                orgs: orgIds,
+              });
+            }
+          });
+
+        layerVersion.legacyLayer = layerVersion.LogicalName === undefined || layerVersion.LogicalName === 'LambdaLayer';
+      });
+      layerMetadata = layerVersionList.sort((a: LayerVersionMetadata, b: LayerVersionMetadata) => b.Version - a.Version);
+    } catch (e) {
+      spinner.fail();
+      const errMessage = `An error occurred fetching the latest layer version metadata for "${layerName}": ${e.message || e}`;
+      context.print.error(errMessage);
+      await context.usageData.emitError(new Error(errMessage));
+      exitOnNextTick(1);
+    }
+    spinner.stop();
+    return layerMetadata;
+  }
+
+  public async getLayerVersionsFromCloud(context: $TSContext, layerName: string): Promise<LayerVersionMetadata[]> {
+    if (this.layerVersionsMetadata) {
+      return this.layerVersionsMetadata;
+    }
+    return this.loadLayerDataFromCloud(context, layerName);
```

Comment:
`return this.layerVersionsMetadata || this.loadLayerDataFromCloud(context, layerName);`